dev/service_now/lib/support_ticket_service_now_service.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+# Service class responsible to create a support ticket and delivery it via ServiceNow API
+#
+# It implements the support ticket interface as defined in the SupportTicketController
+class SupportTicketServiceNowService
+  include SupportTicketService
+
+  attr_reader :support_ticket_config
+
+  # Constructor
+  #
+  # @param [Hash] support_ticket_config Support ticket configuration
+  def initialize(support_ticket_config)
+    @support_ticket_config = support_ticket_config
+  end
+
+  # Creates a support ticket in the ServiceNow system configured
+  #
+  # @param [SupportTicket] support_ticket support ticket created in validate_support_ticket
+  #
+  # @return [String] success message
+  def deliver_support_ticket(support_ticket)
+    service_config = support_ticket_config.fetch(:servicenow_api, {})
+    session = get_session(support_ticket)
+    job = get_job(support_ticket)
+    description = create_description_text(service_config, support_ticket, session, job)
+    payload = {
+      caller_id:         support_ticket.email,
+      watch_list:        support_ticket.cc,
+      short_description: support_ticket.subject,
+      description:       description,
+    }
+
+    mapping_fields = service_config.fetch(:map, {}).to_h
+    mapping_fields.each do |snow_field, form_field|
+      # Map field names from the form into field names from ServiceNow
+      # arrays are supported for form_field names and the values will be joined with commas.
+      value = Array.wrap(form_field).map { |name| support_ticket.send(name).to_s }.reject(&:blank?).join(',')
```

Comment:
why use `send(name)` in `support_ticket.send(name).to_s` instead of `support_ticket.name.to_s` ? Is `name` a private method? if it is public, I would use the latter syntax. If it is private and we need to access it, it is ok to keep `send`